CONTRIBUTING.md

```diff
@@ -1,49 +1,122 @@
 # OSML Contributing Guidelines
 
 Thank you for your interest in contributing to our project! This document will guide you through the process of
-contributing to our repository using a release-based GitFlow workflow. Please read the guidelines carefully in order to
+contributing to our repository using a release-based GitFlow workflow. Please read the guidelines carefully to
 submit your pull requests effectively.
 
 ## Table of Contents
 
-- [Release-based GitFlow](#release-based-gitflow)
-- [Branches](#branches)
+- [Trunk-Based GitFlow](#trunk-based-gitflow)
 - [Linting](#linting)
-- [Use Cases](#use-cases)
-   - [Develop a new feature](#develop-a-new-feature)
-   - [Develop multiple features in parallel](#develop-multiple-features-in-parallel)
-   - [Create and deploy a release](#create-and-deploy-a-release)
-   - [Production hot fix](#production-hot-fix)
 - [Code Style](#code-style)
 - [Commit Messages](#commit-messages)
 - [Issue Tracking](#issue-tracking)
 
-## Release-based GitFlow
+## Trunk-Based GitFlow
 
-We follow a release-based GitFlow branching model to manage our codebase. It involves different types of branches to
-ensure proper version control and collaboration. The primary branches in this workflow are:
+![Trunk-Based GitFlow](./images/tbd.png)
 
-note- `main`: This branch represents the latest production-ready state of the project code.
-- `dev`: This branch is used as the integration branch, combining all feature branches for a specific release.
-- `release/vX.Y.Z`: Branches in the format of `release/vX.Y.Z` are used to prepare a specific release.
-- `feature/*`: Feature branches are created for the development of new features or significant enhancements.
-- `hotfix/*`: Hotfix branches are created for bug fixes against production code.
+We follow a trunk-based GitFlow branching model to manage our codebase.
+It involves different types of branches to ensure proper version control and collaboration.
+This guide provides a step-by-step example of implementing a Trunk-Based Development (TBD) workflow,
+covering feature development, rebasing, and the hotfix process.
+The primary branches developers create in this workflow are:
 
+- `main`: This branch represents the latest production-ready state of the project code.
+- `feat/*`: `feat` branches are created for the development of new features or significant enhancements.
+- `fix/*`: `fix` branches are created for bug fixes against production code.
+- `chore/*`: `chore` branches represent simple updates to the code that do not impact functionality.
+- `build/*`: `build` branches represent updates `.github` workflow actions.
 
-## Branches
+---
+**NOTE**
 
-Here's a brief description of the supported branches in our repository:
+Breaking changes would require `!` in front of the prefix (ex: `feat!`)
```

Comment:
Not quite. You can still put a commit with a commit message of `feat!: new API breaking change` on a branch named `feature/implement-new-api`.

I think it would be worthwhile to add a new section on conventional commit messages for this contributor guide.

Reply:
We already have the conventional commits section - so I will just remove this notes section then?